services/chronikService.ts

```diff
@@ -59,73 +143,73 @@ export class ChronikBlockchainClient implements BlockchainClient {
-    } catch (err) {
-      console.log('Error converting address to hash160')
-      throw err
-    }
+export function toHash160 (address: string): {type: ScriptType, hash160: string} {
+  try {
+    // decode address hash
+    const { type, hash } = decode(address)
+    // encode the address hash to legacy format (bitcoin)
+    const legacyAdress = bs58.encode(hash)
+    // convert legacy to hash160
+    const addrHash160 = Buffer.from(bs58.decode(legacyAdress)).toString(
+      'hex'
+    )
+    return { type: type.toLowerCase() as ScriptType, hash160: addrHash160 }
+  } catch (err) {
+    console.log('Error converting address to hash160')
+    throw err
   }
+}
 
-  private outputScriptToAddress (outputScript: String): string | boolean {
-    // returns P2SH or P2PKH address
-    // P2PKH addresses are in outputScript of type 76a914...88ac
-    // P2SH addresses are in outputScript of type a914...87
-    // Return false if cannot determine P2PKH or P2SH address
-
-    const typeTestSlice = outputScript.slice(0, 4)
-    let addressType
-    let hash160
-    switch (typeTestSlice) {
-      case '76a9':
-        addressType = 'P2PKH'
-        hash160 = outputScript.substring(
-          outputScript.indexOf('76a914') + '76a914'.length,
-          outputScript.lastIndexOf('88ac')
-        )
-        break
-      case 'a914':
-        addressType = 'P2SH'
-        hash160 = outputScript.substring(
-          outputScript.indexOf('a914') + 'a914'.length,
-          outputScript.lastIndexOf('87')
-        )
-        break
-      default:
-        return false
-    }
-
-    // Test hash160 for correct length
-    if (hash160.length !== 40) {
+export function outputScriptToAddress (outputScript: String): string | boolean {
+  // returns P2SH or P2PKH address
+  // P2PKH addresses are in outputScript of type 76a914...88ac
+  // P2SH addresses are in outputScript of type a914...87
+  // Return false if cannot determine P2PKH or P2SH address
+
+  const typeTestSlice = outputScript.slice(0, 4)
+  let addressType
+  let hash160
+  switch (typeTestSlice) {
+    case '76a9':
+      addressType = 'P2PKH'
+      hash160 = outputScript.substring(
+        outputScript.indexOf('76a914') + '76a914'.length,
+        outputScript.lastIndexOf('88ac')
+      )
+      break
+    case 'a914':
+      addressType = 'P2SH'
+      hash160 = outputScript.substring(
+        outputScript.indexOf('a914') + 'a914'.length,
+        outputScript.lastIndexOf('87')
+      )
+      break
+    default:
       return false
-    }
-
-    const buffer = Buffer.from(hash160, 'hex')
+  }
 
-    // Because ecashaddrjs only accepts Uint8Array as input type, convert
-    const hash160ArrayBuffer = new ArrayBuffer(buffer.length)
-    const hash160Uint8Array = new Uint8Array(hash160ArrayBuffer)
-    for (let i = 0; i < hash160Uint8Array.length; i += 1) {
-      hash160Uint8Array[i] = buffer[i]
-    }
+  // Test hash160 for correct length
```

Comment:
No need for comments that are just English versions of the code. Better to explain why we're bothering to check it.

Reply:
obvious comments removed